source/jquery.flot.time.js

```diff
@@ -31,11 +31,11 @@ API.txt for details.
 
         var oldSetTime = newDate.setTime.bind(newDate);
         newDate.update = function(microEpoch) {
-            oldSetTime(microEpoch);
-
             // Round epoch to 3 decimal accuracy
             microEpoch = Math.round(microEpoch * 1000) / 1000;
 
+            oldSetTime(microEpoch);
```

Comment:
Can you summarize the need for this particular rearrangement?

Reply:
Thanks for the quick review! I'll try to be as concise as possible but it was not easy to debug, so it's not easy to sum up in a few words 😁

Currently, the rounding of the microEpoch is made **after** forwarding the call to the default `Date.setTime`. And Date.setTime does not round the given value to the nearest millisecond but instead truncates it. Example: `new Date().setTime(0.9999)` returns `0`, not `1`. But in the custom micro epoch implementation, the value is rounded, not truncated. So calling 

I made a simple demo show-casing this behavior here:
current: https://jsfiddle.net/powerhat/o9b8wvy6/4/
after fix: https://jsfiddle.net/powerhat/a3e0rpLo/2/

In addition, this small issue is sometimes causing infinite loops in `dateTickGenerator`.
https://github.com/flot/flot/blob/0b84bfb148922c000768064a2391f2ba4a5d97bc/source/jquery.flot.time.js#L464
Sometimes, by mis-chance, the next tick will be `0.9999` which should be rounded to `1` (nearest micro epoch). Because this value is passed to the micro date `setTime` which has the issue described above and in the jsfiddles, the time is rolled back to `0`. 

Hope that helps